### ISSUE TITLE
USB-C DAC issues

### ISSUE THREAD

**Issue by NadavB9** [GrapheneOS/os-issue-tracker#3728]
Hi, I have issues with my usb-c DAC ever since a few updates ago ( I dont remember which update broke it). Tested on a friend's stock pixel to make sure it's a Gos issue.

1. There is no longer bit perfect playback (meaning its stuck on 48khz) even with a player like tidal or USB audio player pro which bypasses Android audio system.
2.  There is weird noises being passthrough to the DAC when nothing is playing. 
3. there is button presses and other sounds which should not be playing according to stock pixel behavior (when allowing tidal to take over the Dac). 
4. There is random disconnects

I have made sure to change the USB security setting to `ON` instead of `Charging-only when locked`. Again all of those issues do not happen on multiple other phones which I have tested the device on.

- Pixel 7 pro
- Questyle m15i DAC

**Comment by privacyadmin** [comment]
+1 on the USB DAC issue.

Loud distortion can be heard the moment at any volume above silence

**Comment by NadavB9** [comment]
> @NadavB9 are you still have the issues you mentioned?

Yes, still having the exact same issues

**Comment by Vermuden** [comment]
@maade93791 Do you mean the software I use to listen to music? Tidal, but I have the same issue on youtube oder other apps.

**Comment by privacyadmin** [comment]
More than 4 months and the problem still persist on GOS Pixel 9 Pro XL. 

Using the same DAC still working fine on my other phones like unmodified Pixel 7 and iPhone 15 Pro Max..

Even bought a new USB DAC and the problem still the same.

**Comment by privacyadmin** [comment]
A defective DAC can be ruled out since

1: Works on every other phones except GrapheneOS

2: Used to work with GrapheneOS prior. Issue occurs probably somewhere around Jun to Jul period and has been like this ever since.

3. Multiple other brands and models DACs have the same issue.

**Comment by Vermuden** [comment]
I have a similar experience. The DAC works fine with other devices, but with my phone I constantly lose the connection if the phone or DAC moves even a little, or I just look at it wrong. I have the same problem with a different DAC. I have now had these problems on 2 Pixel 8s with both GrapheneOS. It feels like the connection is just very unstable and other devices are much less prone to this.

**Comment by privacyadmin** [comment]
@thestinger I just tested the all my DACs on my spare Pixel 8 Pro and Pixel tablet as well. Let me summarize this in point form.

1: Pixel 8 Pro, Pixel 9 Pro XL and Pixel Tablet are running GOS. 

2: Pixel 7 is running stock Google Android

3. All DACs exhibit the same problem on Pixel 8 Pro and 9 Pro XL.

4: DACs work correctly on Pixel Tablet, Pixel 7 and iPhone

In other words, problem exhibit only on phones running GOS. Tablet running in GOS is working fine with DAC. 

If the insistence is still that my DACs or my Pixel 9 Pro XL and Pixel 8 Pro are also faulty, then I will give up on convincing the team.

I'm not sure what is the difference in USB DAC setup between tablet and phones but I hope this can at least convince you that something is up with the current implementation. 

So in summary, my DACs are working fine. My non-GOS phones are also working fine. My tablet is working fine too.

Only phones running GOS are not.

Thanks for reading.

**Comment by Vermuden** [comment]
@thestinger I did some more tests with the DAC and the same and other devices I have, and only my phone seems to be affected. I tested the DAC before installing GrapheneOS on the same device, and if I remember correctly, the problems did not appear (although I only tested it for a very short time).
Is there any way I can get more information from the phone (e.g. logs specific to the USB C port) to help troubleshoot this?

**Comment by Doc-4** [comment]
It's quite a bit later, but I gave USB DACs a test and was not able to reproduce any issues other than poweramp causing issues when set to "High Res" output (use AAudio instead, just as good), and the haptic noises. Yes, latest Poweramp on latest GOS as of writing doesn't do higher than 48khz, this is more than likely an issue of compatibility between that mode and pixels or an upstream bug. Other apps work fine, I would suggest not using that mode on pixels, has always been temperamental for mine, it's currently more broken than it's been in months. It refuses to pause correctly and refuses to have the volume changed once it started.

The noises you hear at silence are probably just the noise floor or some kind of interference, unless they sound meaningfully different from those? When it's the sole thing plugged into the phone, I don't have much of an audible noise floor. 

My DAC itself seems to work just fine. It's a digital DAC, the official Google one which they apparently stopped making at some point. I believe about six months ago I briefly used a Focusrite Scarlett Solo (full size audio interface), and that worked as well. I have an analog one as well, it obviously doesn't work whatsoever, but I'm saying it to be thorough.

Yes, I've noticed the haptic feedback type noises, but it's not all of them, and they're more muted. I assumed they were intended. I don't have stock to test against.

Outside of that one specific Poweramp mode, the Pixel 8 Pro with GOS has actually had the least amount of problems with wired peripherals of any android I have, and I probably use more than most. A Kernel crash with one external display in one specific mode about 6 months ago is the only hard issue I've had, and that one's likely upstream too.

**Comment by thestinger** [comment]
@Vermuden Are you still having any issues?

**Comment by Vermuden** [comment]
Yes, all the issues I had before are still there. All the hardware is still the same. I'm using the same DAC, headphones and Pixel 8 phone. The DAC works fine on occasion; sometimes, however, the connection is really unstable, and I have to plug it in and out several times. Other times, the connection is lost after a random amount of time. Often, the DAC simply does not power up, and is not recognised by the phone at all. I have not yet found any recurring factor. The DAC works perfectly with my other devices, e.g. laptop.

After plugging it in, the phone's logs show something relating to USB, but I'm not sure what it means.

**Comment by Doc-4** [comment]
I did find a workaround for the buggyness in Poweramp high res mode. Force all outputs under high res to 24bit instead of the autodetected 32 (at least on my pixel 8 pro), it's probably mistakenly picking an unsupported output
